pkg/distro/fedora/images.go

```diff
@@ -174,25 +174,14 @@ func osCustomizations(
 			Compression: true,
 		}
 
-		if oscapConfig.Tailoring != nil {
-			newProfile, tailoringFilepath := oscap.GetTailoringFile(oscapConfig.ProfileID)
-
-			tailoringOptions := osbuild.OscapAutotailorConfig{
-				NewProfile: newProfile,
-				Datastream: datastream,
-				ProfileID:  oscapConfig.ProfileID,
-				Selected:   oscapConfig.Tailoring.Selected,
-				Unselected: oscapConfig.Tailoring.Unselected,
-			}
-
-			osc.OpenSCAPTailorConfig = osbuild.NewOscapAutotailorStageOptions(
-				tailoringFilepath,
-				tailoringOptions,
-			)
-
-			// overwrite the profile id with the new tailoring id
-			oscapStageOptions.ProfileID = newProfile
-			oscapStageOptions.Tailoring = tailoringFilepath
+		osc.OpenSCAPTailorConfig = oscap.CreateTailoringStageOptions(
+			oscapConfig,
+			t.arch.distro,
+		)
+
+		if tailorConfig := osc.OpenSCAPTailorConfig; tailorConfig != nil {
```

Comment:
Is this only to make the following two lines shorter? I must admit that I had to stop for a second to carefully read the code and understand that you are assigning a variable to another variable just to check if it is `nil`...